packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/routers/router.dart

```diff
@@ -1,17 +1,232 @@
+    TypedGoRoute<EnterMessageRoute>(path: 'enterMessage'),
+    TypedGoRoute<ItemsRoute>(path: 'items', routes: <TypedGoRoute<GoRouteData>>[
+      TypedGoRoute<ItemDetailsRoute>(path: ':id'),
+    ]),
   ],
 )
-class $Router {}
+@immutable
+class CounterRoute extends GoRouteData implements RouteData {
+  const CounterRoute();
+
+  @override
+  Page<Function> buildPage(BuildContext context, GoRouterState state) =>
+      MaterialPage(
+        key: state.pageKey,
+        child: const CounterPageWithDependencies(),
+      );
+
+  @override
+  String get permissionName => RouterPermissionNames.counterPage;
+
+  @override
+  String get routeLocation => location;
+}
+
+@immutable
+class NotificationsRoute extends GoRouteData implements RouteData {
+  const NotificationsRoute();
+
+  @override
+  Page<Function> buildPage(BuildContext context, GoRouterState state) =>
+      MaterialPage(
+        key: state.pageKey,
+        child: const NotificationsPageWithDependencies(),
+      );
+
+  @override
+  String get permissionName => RouterPermissionNames.notificationsPage;
+
+  @override
+  String get routeLocation => location;
+}
+
+@immutable
+class LoginRoute extends GoRouteData implements RouteData {
+  const LoginRoute();
+
+  @override
+  Page<Function> buildPage(BuildContext context, GoRouterState state) =>
+      MaterialPage(
+        key: state.pageKey,
+        child: const LoginPageWithDependencies(),
+      );
+
+  @override
+  String get permissionName => RouterPermissionNames.loginPage;
+
+  @override
+  String get routeLocation => location;
+}
+
+@immutable
+class EnterMessageRoute extends GoRouteData implements RouteData {
+  const EnterMessageRoute();
+
+  @override
+  Page<Function> buildPage(BuildContext context, GoRouterState state) =>
+      MaterialPage(
+        key: state.pageKey,
+        child: const EnterMessageWithDependencies(),
+      );
+
+  @override
+  String get permissionName => RouterPermissionNames.enterMessagePage;
+
+  @override
+  String get routeLocation => location;
+}
+
+@immutable
+class ItemsRoute extends GoRouteData implements RouteData {
+  const ItemsRoute();
+
+  @override
+  Page<Function> buildPage(BuildContext context, GoRouterState state) =>
+      MaterialPage(
+        key: state.pageKey,
+        child: const ItemsListWithDependencies(),
+      );
+
+  @override
+  String get permissionName => RouterPermissionNames.itemsPage;
+
+  @override
+  String get routeLocation => location;
+}
+
+@immutable
+class ItemDetailsRoute extends GoRouteData implements RouteData {
+  const ItemDetailsRoute(this.id);
+
+  final String id;
+
+  @override
+  Page<Function> buildPage(BuildContext context, GoRouterState state) {
+    ItemModel? item;
+    if (state.extra != null && state.extra is ItemModel) {
+      item = (state.extra as ItemModel);
+    }
+    return MaterialPage(
+      key: state.pageKey,
+      child: ItemDetailsWithDependencies(
+        itemId: id,
+        item: item,
```

Comment:
Can't squash this to just `state.extra as? ItemModel`

Reply:
Done